### ISSUE TITLE
Decommission fails and gets blocked on the failed compaction on the Scylla node that was restarted

### ISSUE THREAD

**Issue by vponomaryov** [scylladb/scylladb#11976]
## Installation details

Kernel Version: 5.4.219-126.411.amzn2.x86_64
Scylla version (or git commit hash): `5.2.0~dev-20221112.ff87624fb4d6` with build-id `faf25f6ca5e5efa4714b3eef769a784b1d76ffc4`

Cluster size: 3 nodes (`i3.4xlarge`)

Scylla Nodes used in this run:
- sct-cluster-us-east1-b-us-east0
- sct-cluster-us-east1-b-us-east1
- sct-cluster-us-east1-b-us-east2

OS / Image: `` (k8s-eks: eu-north-1)

Test: `longevity-scylla-operator-basic-12h-eks`
Test id: `111aaa43-4fd3-4096-bbbe-213b92683452`
Test name: `scylla-operator/operator-master/eks/longevity-scylla-operator-basic-12h-eks`
Test config file(s):

- [longevity-scylla-operator-basic-12h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/edd40ec9434202317c397afe574df7f057147341/test-cases/scylla-operator/longevity-scylla-operator-basic-12h.yaml)

## Issue description

Running on the K8S platform we run `terminate_decommission_add_node_kubernetes` nemesis, among all other, with the following scenario:
1) Drain K8S node (Stop Scylla pod/node completely) - it also inclused "cordon K8S node" action which makes it be unschedulable
2) Delete stopped scylla pod/node data
3) Uncordon K8S node (make it be schedulable)
4) Scylla pod gets started without data but with the same membership
5) Decommission restarted Scylla pod/node
6) Terminate K8S node (Autoscaler will recreate it)
7) Create new Scylla pod/node

And with the latest master Scylla build the step `5` is no longer working. That restarted pod/node falls into endless loop of the following kind of the error:

```
ERROR 2022-11-13 07:28:30,028 [shard  0] system_keyspace - update compaction history failed: exceptions::mutation_write_failure_exception (Operation failed for system.compaction_history - received 0 responses and 1 failures from 1 CL=ONE.): ignored
...
340729 ERROR 2022-11-13 07:28:30,043 [shard  0] init - Startup failed: std::bad_cast (std::bad_cast)       
340730 2022-11-13 07:28:30,118 INFO exited: scylla (exit status 1; not expected)                           
340731 2022-11-13 07:28:31,121 INFO spawned: 'scylla' with pid 221                                         
340732 Scylla version 5.2.0~dev-0.20221112.ff87624fb4d6 with build-id faf25f6ca5e5efa4714b3eef769a784b1d76ffc4 starting ...
...
343288 INFO  2022-11-13 07:28:33,637 [shard  0] compaction_manager - Stopping 1 tasks for 1 ongoing compactions for table system.truncated due to table removal
343289 INFO  2022-11-13 07:28:33,640 [shard  0] compaction - [Compact system.truncated c78ecb30-6324-11ed-81ac-153dfe746a67] Compacting of 7 sstables interrupted due to: sstables::compaction_stopped_exception (Compaction for system/truncated was stopped due to: table removal)
343290 INFO  2022-11-13 07:28:33,640 [shard  0] compaction_manager - Compaction task 0x6000000a6b10 for table system.truncated [0x6000009f5e40]: Compaction for system/truncated was stopped due to: table removal: stopping
...
343396 ERROR 2022-11-13 07:28:33,690 [shard  0] init - Startup failed: std::bad_cast (std::bad_cast)       
343397 2022-11-13 07:28:33,766 INFO exited: scylla (exit status 1; not expected)                           
343398 2022-11-13 07:28:34,769 INFO spawned: 'scylla' with pid 259                                         
343399 Scylla version 5.2.0~dev-0.20221112.ff87624fb4d6 with build-id faf25f6ca5e5efa4714b3eef769a784b1d76ffc4 starting ...
```

So, the node never gets decommissioned after it.

- Restore Monitor Stack command: `$ hydra investigate show-monitor 111aaa43-4fd3-4096-bbbe-213b92683452`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=111aaa43-4fd3-4096-bbbe-213b92683452)
- Show all stored logs command: `$ hydra investigate show-logs 111aaa43-4fd3-4096-bbbe-213b92683452`

## Logs:
- **db-cluster-111aaa43.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/db-cluster-111aaa43.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/db-cluster-111aaa43.tar.gz)
- **monitor-set-111aaa43.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/monitor-set-111aaa43.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/monitor-set-111aaa43.tar.gz)
- **loader-set-111aaa43.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/loader-set-111aaa43.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/loader-set-111aaa43.tar.gz)
- **sct-runner-111aaa43.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/sct-runner-111aaa43.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/sct-runner-111aaa43.tar.gz)
- **kubernetes-111aaa43.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/kubernetes-111aaa43.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/111aaa43-4fd3-4096-bbbe-213b92683452/20221113_090244/kubernetes-111aaa43.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-operator/job/operator-master/job/eks/job/longevity-scylla-operator-basic-12h-eks/33/)

**Comment by fruch** [comment]
seems like this `bad_cast` appeared recently in other runs as well:
https://github.com/scylladb/scylladb/issues/11966

**Comment by DoronArazii** [comment]
@bhalevy / @raphaelsc please have a look

**Comment by mykaul** [comment]
@vponomaryov - the following step is unclear to me "Delete stopped scylla pod/node data"  - why would that happen? what's the use case here?

**Comment by fruch** [comment]
> @vponomaryov - the following step is unclear to me "Delete stopped scylla pod/node data" - why would that happen? what's the use case here?

This simulate the case for k8s node going down, and this pod need to schedule to a new k8s node (where it doesn't have it's pv)
it's a poorsman substitution for really taking the k8s node down (we still have that on our backlog to implant)

anyhow the issue here seems to be unrelated to k8s/operator, since it's seems to be happen in VMs as well

**Comment by vponomaryov** [comment]
> > @vponomaryov - the following step is unclear to me "Delete stopped scylla pod/node data" - why would that happen? what's the use case here?
> 
> This simulate the case for k8s node going down, and this pod need to schedule to a new k8s node (where it doesn't have it's pv) it's a poorsman substitution for really taking the k8s node down (we still have that on our backlog to implant)
> 
> anyhow the issue here seems to be unrelated to k8s/operator, since it's seems to be happen in VMs as well

Israel is right, it is simulation of a pretty real situation where we lose a K8S node and all it's storage.
And, as Israel also stated, it is most probably unrelated to K8S and is one another indicator of a bug that occurred here: https://github.com/scylladb/scylladb/issues/11966

**Comment by mykaul** [comment]
This bug seems like an operator issue to me - the operator should not fail next node before the previous node has fully rejoined the cluster.

**Comment by vponomaryov** [comment]
> This bug seems like an operator issue to me - the operator should not fail next node before the previous node has fully rejoined the cluster.

Which `next` node? In this scenario only one Scylla node gets operated on.

**Comment by DoronArazii** [comment]
@raphaelsc / @bhalevy any updates here?

**Comment by raphaelsc** [comment]
> @raphaelsc / @bhalevy any updates here?

Not yet. Will look into it today.

**Comment by raphaelsc** [comment]
340729 ERROR 2022-11-13 07:28:30,043 [shard  0] init - Startup failed: std::bad_cast (std::bad_cast)       
340730 2022-11-13 07:28:30,118 INFO exited: scylla (exit status 1; not expected)       

@bhalevy does this ring a bell to you? Seems like the problem Aleksandra fixed.

**Comment by raphaelsc** [comment]
Fixed in commit https://github.com/scylladb/scylladb/commit/dcc17037c77eeec4e1e301a15e3b66eccb776d80

**Comment by bhalevy** [comment]
So is it a duplicate of https://github.com/scylladb/scylladb/issues/11966?

**Comment by raphaelsc** [comment]
> So is it a duplicate of #11966?

Looks like so to me. Manifests exactly the same way.
